test_unstructured_ingest/test-ingest-azure.sh

```diff
@@ -8,6 +8,19 @@ OUTPUT_FOLDER_NAME=azure
 OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
 DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
 
+function cleanup() {
+  echo "--- Running cleanup ---"
+
+  if [ -d "$OUTPUT_DIR" ]; then
+    echo "cleaning up tmp directory: $OUTPUT_DIR"
+    rm -rf "$OUTPUT_DIR"
+  fi
+
+  echo "--- Cleanup done ---"
+}
+
+trap cleanup EXIT
+
```

Comment:
can we make this a common script that just gets called rather than so much duplication?